src/coreclr/jit/lowerarmarch.cpp

```diff
@@ -2264,7 +2264,30 @@ void Lowering::ContainCheckCast(GenTreeCast* node)
 //
 void Lowering::ContainCheckCompare(GenTreeOp* cmp)
 {
-    CheckImmedAndMakeContained(cmp, cmp->gtOp2);
+    GenTree* op1 = cmp->gtGetOp1();
+    GenTree* op2 = cmp->gtGetOp2();
+
+    if (CheckImmedAndMakeContained(cmp, op2))
+        return;
```

Comment:
You can also handle `op1` being an immediate and doing the relop swap here.

Something like:
```cpp
    if (CheckImmedAndMakeContained(cmp, op1))
    {
        std::swap(cmp->gtOp1, cmp->gtOp2);
        cmp->ChangeOper(cmp->SwapRelop(cmp->gtOper));
        return;
    }
```